src/plugin.js

```diff
@@ -101,6 +102,13 @@ const initPlugin = function(player, options) {
         return;
       }
 
+      // if the player has started to stall and
+      // playback is almost at the end just end playback
+      if ((player.duration() - player.currentTime()) < options.durationTolerance) {
```

Comment:
It's a rare case, but I believe it is possible that in HLS, if we start the video the manifest hasn't downloaded, and we can't get a source buffer (or it takes a long time for some reason), this will trigger ended after we get waiting (since duration will return as 0 and current time as 0). Although it is rare, and we can probably resolve it, we may just want to check that duration is > 0 to be safe. I believe the standard for media sources is NaN, which should be fine, but, since different technologies may follow different behaviors when returning a non-existent duration, we may want to have the extra check.